pkg/controller/nodenetworkconfigurationpolicy/nodenetworkconfigurationpolicy_controller.go

```diff
@@ -180,6 +191,9 @@ func (r *ReconcileNodeNetworkConfigurationPolicy) Reconcile(request reconcile.Re
 
 	conditionsManager.NotifyMatching()
 
+	unlock := r.rollout.Lock()
+	defer unlock()
```

Comment:
We don't want to unlock at the end `Reconcile` we want to unlock at the end of apply so we update status in parallel.

Reply:
In other words, call the `unlock` right after `ApplyDesiredState` line.